crates/driver/src/domain/competition/bad_tokens/metrics.rs

```diff
@@ -1,11 +1,67 @@
-use {super::Quality, crate::domain::eth};
+use {super::Quality, crate::domain::eth, dashmap::DashMap, std::sync::Arc};
 
+/// Monitors tokens to determine whether they are considered "unsupported" based
+/// on the ratio of failing to total settlement encoding attempts. A token must
+/// have participated in at least `REQUIRED_MEASUREMENTS` attempts to be
+/// evaluated. If, at that point, the ratio of failures is greater than or equal
+/// to `FAILURE_RATIO`, the token is considered unsupported.
 #[derive(Default)]
-pub struct Detector;
+pub struct Detector(Arc<Inner>);
+
+#[derive(Default)]
+struct Inner {
```

Comment:
Alternatively this could be changed to type alias or `DashMap` could be directly used in `Arc<>`, so we could skip accessing by `.0.counter`.

Is there any special reason for using `Inner` here? Possible future extensions of that struct?

Reply:
The reason is to make the cache private.

Reply:
The detector thresholds should be configurable so having a separate struct makes sense here.

Reply:
Ah, ok. The structure is now extended to a separate PR.